core/sql/parser/sqlparser.y

```diff
@@ -33167,7 +33173,106 @@ stats_merge_clause : /*empty */
                { $$ = SQLCLI_PROGRESS_STATS; }
              | TOK_DEFAULT
                { $$ = SQLCLI_SAME_STATS; }  
-                   
+
+
+/* type pStmtDDL */
+comment_on_statement : TOK_COMMENT TOK_ON comment_on_object_types ddl_qualified_name TOK_IS QUOTED_STRING
+               {
+                 StmtDDLCommentOn *pNode = 
+                   new(PARSERHEAP()) StmtDDLCommentOn(
+                     $3,
+                     *$4,
+                     *$6,
+                     PARSERHEAP()
+                   );
+
+                 $$ = pNode;
+               }
+             | TOK_COMMENT TOK_ON TOK_SCHEMA schema_name TOK_IS QUOTED_STRING
+               {
+                 // EJF L4J - CQD dynamic are not allowed in Compound Statements
```

Comment:
The comment is incorrect. But was it your intent to disallow COMMENT ON within a compound statement? If so, you may want to disallow it in the first TOK_COMMENT production as well.

Reply:
No, i do not want to disallow COMMENT ON within a compound statement.
I will remove this comment and if branch.